fastcache.go

```diff
@@ -413,3 +418,30 @@ func (b *bucket) Del(h uint64) {
 	delete(b.m, h)
 	b.mu.Unlock()
 }
+
+func (b *bucket) copyKeys() ([][]byte, int) {
+	b.mu.RLock()
+
+	keys := make([][]byte, len(b.m))
+	numKeys := 0
+
+	for _, v := range b.m {
+		idx := v & ((1 << bucketSizeBits) - 1)
+		gen := v >> bucketSizeBits
+
+		if gen == b.gen && idx < b.idx || gen+1 == b.gen && idx >= b.idx {
+			chunkIdx := idx / chunkSize
+			chunk := b.chunks[chunkIdx]
+
+			kvLenBuf := chunk[idx : idx+4]
```

Comment:
This may cause the array to go out of bounds and generate Panic

Reply:
Adding this line of code may solve the problem, can you help me take a look?  idx = idx % chunkSize